std/random.d

```diff
@@ -1183,6 +1183,135 @@ A singleton instance of the default random number generator
 }
 
 /**
+A random bool generator that has the underlying random integral
+generation batched. The underlying random number generator can be
+passed as a pointer so the RNG will not get copied internally.
+ */
+struct RandomBool(Rng = Random)
+    if (isUniformRNG!Rng || isUniformRNG!(typeof(*Rng.init)))
+{
+    private Rng parent;
+    private uint mask;
+    private uint accumulator;
+
+    /// Always `false`
+    enum bool empty = false;
+
+    /**
+      If Rng is not a pointer, we want to pass it by ref
+      in order to avoid two passByValues
+     */
+    this()(auto ref Rng rng)
+    {
+        parent = rng;
+        initPrivates();
+    }
+
+    private void initPrivates()
+    {
+        // Helper function to avoid code duplication
+        mask = 1;
+        accumulator = uniform!uint(parent);
+    }
+
+    bool front()
+    {
+        return cast(bool)(accumulator & mask);
+    }
+
+    void popFront()
+    {
+        if (mask == 0)
+        {
+            initPrivates();
+        }
+
+        mask <<= 1;
```

Comment:
I have made the requested changes. I also did a rebase in order to have this branch in line with master.